mongo/mongo.go

```diff
@@ -90,8 +92,9 @@ func (m *Mongo) GetCacheTime(ctx context.Context, id string) (*models.CacheTime,
 
 	err := m.Connection.Collection(m.ActualCollectionName(config.CacheTimesCollection)).FindOne(ctx, filter, &result)
 	if err != nil {
-		// TODO: need to differentiate between 404 not found err and internal server error.
-		log.Error(ctx, "error finding dataset", err)
```

Comment:
The logging shouldn't have been removed, in fact there should probably be an extra logging line now (one for the not found case, another one for the regular error case).

Reply:
We are following the convention set out in a similar dp-dataset-api method here: https://github.com/ONSdigital/dp-dataset-api/blob/develop/mongo/dataset_store.go#L77. Unless I'm wrong, I thought the error caught here would bubble up to the handler, so it would be logged in the handler too in the case that 404 not found is raised here? Would implementing it here mean that the log was output twice?

Reply:
That's alright then. It just means that you need to remember to do the logging everywhere you're using this function, but it looks like we're only planning on using this function in one place, so I guess it should be fine.

Reply:
this should now be addressed due to changes in how 500 errors were handled